src/main/scala/no/ndla/draftapi/controller/UserDataController.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Part of NDLA draft-api.
+ * Copyright (C) 2020 NDLA
+ *
+ * See LICENSE
+ */
+
+package no.ndla.draftapi.controller
+
+import no.ndla.draftapi.auth.User
+import no.ndla.draftapi.model.api.{UserData, UpdatedUserData}
+import no.ndla.draftapi.service.{ReadService, WriteService}
+import org.scalatra.Ok
+import org.scalatra.swagger.{ResponseMessage, Swagger}
+
+import scala.util.{Failure, Success}
+
+trait UserDataController {
+  this: ReadService with WriteService with User =>
+  val userDataController: UserDataController
+
+  class UserDataController(implicit val swagger: Swagger) extends NdlaController {
+    protected val applicationDescription = "API for accessing user data."
+    val response400: ResponseMessage = ResponseMessage(400, "Validation Error", Some("ValidationError"))
+    val response403: ResponseMessage = ResponseMessage(403, "Access Denied", Some("Error"))
+    val response404: ResponseMessage = ResponseMessage(404, "Not found", Some("Error"))
+    val response500: ResponseMessage = ResponseMessage(500, "Unknown error", Some("Error"))
+
+    private val query =
+      Param[Option[String]]("query", "Return only user data with content matching the specified query.")
+
+    get(
+      "/",
+      operation(
+        apiOperation[UserData]("getUserData")
+          summary "Retrieves user's data"
+          description "Retrieves user's data"
+          parameters (
+            asHeaderParam(correlationId),
+            asQueryParam(query)
```

Comment:
Kva for en parameter er dette? Brukere må kun få tilgang til sine egne user-data og ikkje andre sine.